src/crypto-providers/ledgerCryptoProvider.ts

```diff
@@ -1503,17 +1516,32 @@ export const LedgerCryptoProvider: (
         addressFieldType: LedgerTypes.MessageAddressFieldType.KEY_HASH,
       }
     }
-    try {
-      const response = await ledger.signMessage(ledgerArgs)
 
-      return {
-        signatureHex: response.signatureHex,
-        signingPublicKeyHex: response.signingPublicKeyHex,
-        addressFieldHex: response.addressFieldHex,
-      } as SignedMessageData
+    let response: LedgerTypes.SignMessageResponse
+    try {
+      response = await ledger.signMessage(ledgerArgs)
     } catch (err) {
       throw Error(failedMsg(err))
     }
+
+    if (args.address !== undefined) {
```

Comment:
If `args.address` is undefined, we can check that the returned address field is the hash of the signing key (the key is included in the signing file?).

Reply:
The same check is performed by the signing key check below (I believe the addressField and signingPubKey have the same value in that case?).